app/services/preserved_object_handler.rb

```diff
@@ -163,25 +195,29 @@ def create_db_objects(status, validated=false)
     results.concat(create_results)
   end
 
-  def update_online_version
+  # TODO: this is "too complex" per rubocop: shameless green implementation
+  # NOTE: if reduce complexity, remove Metrics/PerceivedComplexity exception in .rubocop.yml
+  def update_online_version(validated=false, status=nil)
     results = []
     upd_results = with_active_record_transaction_and_rescue do
       pres_object = PreservedObject.find_by!(druid: druid)
-      pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint)
+      pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
       # FIXME: what if there is more than one associated pres_copy?
       if incoming_version > pres_copy.version && pres_copy.version == pres_object.current_version
         results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_copy.class.name)
-        update_preserved_copy(pres_copy, incoming_version, incoming_size)
-        results.concat(update_status(pres_copy, Status.ok))
+        update_preserved_copy_version_etc(pres_copy, incoming_version, incoming_size, validated)
+        results.concat(update_status(pres_copy, status)) if status
         results.concat(update_db_object(pres_copy))
         results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_object.class.name)
-        update_preserved_object(pres_object, incoming_version)
+        pres_object.current_version = incoming_version
```

Comment:
what about also calling `update_pc_validation_timestamps(pres_copy) if validated` here?

Reply:
nah, that's fine, i missed that, and similar concerns here, but no suggestions for fixing them jump to mind